### ISSUE TITLE
-m parameter for 'run.py edges' method not behaving as expected?

### ISSUE THREAD

**Issue by realmarcin** [Knowledge-Graph-Hub/kg-covid-19#217]
### Describe the bug

the difference between -m 20 and -m 1000 was over 2 million edges, the higher degree parameter gave more edges which was surprising. My assumption was that since nodes below a certain degree were not allowed, then there would be fewer nodes to chose from and hence fewer total edges to choose from.

### To Reproduce

 python run.py edges -n data/merged/merged-kg_nodes.tsv -e data/merged/merged-kg_edges.tsv -m 20

wc pos_train_edges.tsv
  13180781   52723124 1085707551 pos_train_edges.tsv

 python run.py edges -n data/merged/merged-kg_nodes.tsv -e data/merged/merged-kg_edges.tsv -m 1000

wc pos_train_edges.tsv 
  15800739   63202956 1300088893 pos_train_edges.tsv

python run.py edges -n data/merged/merged-kg_nodes.tsv -e data/merged/merged-kg_edges.tsv -m 10000

wc pos_train_edges.tsv
  16307600   65230400 1341653621 pos_train_edges.tsv

### Version

Which version of KG-COVID-19 are you seeing the bug with? An md5 hash is most useful.

**Comment by justaddcoffee** [comment]
I believe what you are seeing is the expected behavior. 

The `-m` parameter imposes a minimum degree for edges to be selected for the test (and optionally the validation) set. (Recall the intention of `-m` is to avoid selecting edges for the test edges that are not well-connected, since these edges are removed from the training graph - less well-connected edges are more likely to create new components.) Those edges that are not in the test or validation set are by definition in the training set. When you set `-m` to be very, very high, there are not enough edges that satisfy this condition, so the test set is smaller and the training set is bigger compared with lower values of `-m`.

Another comment - definitely don't set `-m` nearly as high as 1,000 for use in an actual experiment - this will be a very strange subset of edges. (I realize though you probably were just doing these high `-m` runs as a test.)
